app/jobs/appointment_notification/schedule_experiment_reminders.rb

```diff
@@ -0,0 +1,14 @@
+class AppointmentNotification::ScheduleExperimentReminders < ApplicationJob
+  queue_as :high
+
+  def perform
+    return unless Flipper.enabled?(:appointment_reminders)
+
+    reminders = AppointmentReminder.due_today
+    next_messaging_time = Communication.next_messaging_time
+    reminders.each do |reminder|
+      AppointmentNotification::Worker.perform_at(next_messaging_time, reminder.id)
+      reminder.status_scheduled!
```

Comment:
We should probably set the state to `scheduled` first, otherwise there is a race condition where the job picks up the reminder before the status has been set.

Reply:
That's a fair point. I didn't consider it because I expect this script to run hours before the notifications send, because that's how the current notifications work. But tbh I think it's worth considering changing that behavior. I don't see much advantage to scheduling the notifications several hours early. It may have some small performance impact but it could also leave us open to sidekiq/redis errors.